### PR TITLE
fix: set language using url param in language dropdown

### PR DESCRIPTION
**Pull request by justindomingue**

### CODE REVIEW

File: src/components/Menu/index.tsx [hunk 1]
```diff
@@ -163,6 +181,39 @@ const ToggleMenuItem = styled.button`
...
+  const location = useLocation()
+  const qs = useParsedQueryString()
+
+  return (
+    <MenuFlyout>
+      <ToggleMenuItem onClick={close}>
+        <ChevronLeft size={16} />
+      </ToggleMenuItem>
+      {SUPPORTED_LOCALES.map((locale) => (
+        <InternalLinkMenuItem
```
Comment (moodysalem) [inline src/components/Menu/index.tsx:195]: you could put this into a shared component like `SwitchLocaleLink<{ locale: SupportedLocale }>`

Comment (moodysalem) [reply src/components/Menu/index.tsx:195]: or write a hook `useLocationLinkProps(locale: SupportedLocale): { to: Location, onClick: () => void }`

Comment (justindomingue) [reply src/components/Menu/index.tsx]: refactored. Did I overcomplicate the hooks?

File: src/hooks/useLocationLinkProps.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  to?: LocationDescriptor
+  onClick?: () => void
+} {
+  const location = useLocation()
+  const qs = useParsedQueryString()
+  const activeLocale = useActiveLocale()
+
+  if (!locale) return {}
+
+  return {
```
Comment (moodysalem) [inline src/hooks/useLocationLinkProps.ts:19]: memo this whole thing

File: src/components/SwitchLocaleLink/index.tsx [hunk 3]
```diff
@@ -17,46 +14,37 @@ const Container = styled(TYPE.small)`
...
+
+  const { to, onClick } = useLocationLinkProps(targetLocale)
+
+  if (!targetLocale || !to) return null
+
+  return (
+    <Container>
+      <Trans>
+        Uniswap available in:{' '}
+        {
```
Comment (moodysalem) [inline src/components/SwitchLocaleLink/index.tsx:42]: unnecessary bracket

File: src/components/SwitchLocaleLink/index.tsx [hunk 4]
```diff
@@ -17,46 +14,37 @@ const Container = styled(TYPE.small)`
   margin-top: 1rem !important;
 `
 
-export function SwitchLocaleLink() {
-  const activeLocale = useActiveLocale()
+const useTargetLocale = (activeLocale: SupportedLocale) => {
```
Comment (moodysalem) [inline src/components/SwitchLocaleLink/index.tsx:17]: if this werent a hook then target locale would never be null i think and then your hook can always return a value w/ to and onClick

Comment (justindomingue) [reply src/components/SwitchLocaleLink/index.tsx:17]: `targetLocale` is actually conditionally set based on active locale and default locale so I think `null` is a valid state here.

**Review by moodysalem — APPROVED** [review]

**Pull request merged**